File: Data Structures & Algorithms/is-anagram/submission-7.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        charCount = [0] * 26
        for c in s:
            charCount[ord(c) - ord('a')] += 1

        for c in t:
            if charCount[ord(c) - ord('a')] == 0:
                return False
            charCount[ord(c) - ord('a')] -= 1

        return sum(charCount) == 0

File: Data Structures & Algorithms/anagram-groups/submission-7.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> groups;

        for (const string& s : strs) {
            vector<int> charCount(26, 0);
            for (const char& c : s) {
                charCount[c - 'a']++;
            }
            string key = "";
            for (int n : charCount) {
                key += to_string(n);
                key += ",";
            }
            groups[key].push_back(s);
        }

        vector<vector<string>> res;
        for (auto [_, group] : groups) {
            res.push_back(group);
        }

        return res;
    }
};
